server/internal/client/ollama/registry.go

```diff
@@ -514,10 +514,41 @@ func (r *Registry) Pull(ctx context.Context, name string) error {
 				break
 			}
 
+			cacheKey := fmt.Sprintf(
+				"v1 pull chunksum %s %s %d-%d",
+				l.Digest,
+				cs.Digest,
+				cs.Chunk.Start,
+				cs.Chunk.End,
+			)
+			cacheKeyDigest := blob.DigestFromBytes(cacheKey)
+			_, err := c.Get(cacheKeyDigest)
+			if err == nil {
+				// Chunk already downloaded.
+				//
+				// NOTE: This is not the right situation for
+				// ErrCached. It should only look like a fast
+				// chunk download, so pass a nil error.
+				t.update(l, cs.Chunk.Size(), nil)
```

Comment:
Also update received. Currently any completion after any abort and resume will get an "incomplete" error because this does not update received.